### PR TITLE
sv-lang: init at 3.0

### PR DESCRIPTION
**Pull request by SharzyL**
###### Description of changes

sv-lang (also known as slang) provides a SystemVerilog compiler and language services.

Homepage: https://github.com/MikePopoloski/slang

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SharzyL** [comment]
Seems like darwin build failed, looking into it

File: pkgs/applications/science/electronics/sv-lang/default.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}:
+
+let
+  # dependency for this package has been removed in master (i.e. next release)
+  unordered_dense = stdenv.mkDerivation rec {
+    version = "2.0.1";
+    pname = "unordered_dense";
+    src = fetchFromGitHub {
+      owner = "martinus";
+      repo = pname;
```
Comment (sequencer) [inline pkgs/applications/science/electronics/sv-lang/default.nix:16]: Shoulde be "slang"

File: pkgs/applications/science/electronics/sv-lang/default.nix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+in
+stdenv.mkDerivation rec {
+  # its another name `slang` coincides with https://www.jedsoft.org/slang/
+  pname = "sv-lang";
+  version = "3.0";
+
+  src = fetchFromGitHub {
+    owner = "MikePopoloski";
+    repo = pname;
```
Comment (sequencer) [inline pkgs/applications/science/electronics/sv-lang/default.nix:33]:
```suggestion
    repo = "slang";
```

File: pkgs/applications/science/electronics/sv-lang/default.nix [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      sha256 = "sha256-9zlWYAY4lOQsL9+MYukqavBi5k96FvglRgznLIwwRyw=";
+    };
+    nativeBuildInputs = [
+      cmake
+    ];
+  };
+
+in
+stdenv.mkDerivation rec {
+  # its another name `slang` coincides with https://www.jedsoft.org/slang/
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/sv-lang/default.nix:27]:
```suggestion
```

File: pkgs/applications/science/electronics/sv-lang/default.nix [hunk 4]
```diff
@@ -0,0 +1,69 @@
+{ lib, stdenv, fetchFromGitHub
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/sv-lang/default.nix:1]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
```

File: pkgs/applications/science/electronics/sv-lang/default.nix [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+stdenv.mkDerivation rec {
+  # its another name `slang` coincides with https://www.jedsoft.org/slang/
+  pname = "sv-lang";
+  version = "3.0";
+
+  src = fetchFromGitHub {
+    owner = "MikePopoloski";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-v2sStvukLFMRXGeATxvizmnwEPDE4kwnS06n+37OrJA=";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/sv-lang/default.nix:35]:
```
building '/nix/store/d8riws552a8swismca9xanipd1xd9hxn-source.drv'...
building '/nix/store/c26cpidg8828r3xfcaixkdynz3hci19i-source.drv'...
trying https://github.com/martinus/unordered_dense/archive/v2.0.1.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
trying https://github.com/MikePopoloski/sv-lang/archive/v3.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     9    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
error: cannot download source from any mirror
error: builder for '/nix/store/d8riws552a8swismca9xanipd1xd9hxn-source.drv' failed with exit code 1;
```

File: pkgs/applications/science/electronics/sv-lang/default.nix (hunk 3 above)
Comment (SharzyL) [reply pkgs/applications/science/electronics/sv-lang/default.nix]: May I ask where is the change? Two lines look identical.

**Comment by SharzyL** [comment]
Compilation error occurs due to incompatibility between fmt_10 and fmt_9. Fix with switching to fmt_9.

**Comment by SharzyL** [comment]
Linker error on darwin suppressed by disabling tests.

File: pkgs/applications/science/electronics/sv-lang/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/applications/science/electronics/sv-lang/default.nix]: The change is the removal of the comment. pname's can clash, that's not a problem

**Pull request merged**